### ISSUE TITLE
Mapper isn't responding

### ISSUE THREAD

**Issue by erummas** [Health-Informatics-UoN/carrot-mapper#761]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hi, this has happened twice today that the mapper has stopped responding. At first I thought it may be due to some server issues but I think it happens when the concepts are added manually too quickly (I could be wrong) as both times it happened when I was entering the same concept_id for multiple codes so copy pasting quickly across various code values. Both times I am unable to refresh the application. 

### Environment

```markdown
- OS:
- Other environment details:
```

### Are you willing to contribute to resolve this issue?

None

**Comment by AndrewThien** [comment]
Hi @erummas,
Thanks for the reporting.
I have been trying to reproduce the issue on my local machine and the dev environment, but it seems like I can't do that.
Would you mind having a quick video call for me to better understand the problem? Thanks a lot!

**Comment by AndrewThien** [comment]
Hi @erummas , for now I have two videos that maybe related to the issue you mentioned
Video 1: 

https://github.com/Health-Informatics-UoN/Carrot-Mapper/assets/119530400/586a3546-aea5-497b-9df9-ac61a5abda56

Video 2:

https://github.com/Health-Informatics-UoN/Carrot-Mapper/assets/119530400/ec511603-9bfa-426d-a49e-32d88f5832a3

Can you please confirm if there are any of them similar to the issues you reported? Thanks a lot

**Comment by erummas** [comment]
@AndrewThien scenario 1. Haven't tried scenario 2 so can't say how that would respond.

**Comment by erummas** [comment]
I did it the second way too and the issue persists.

**Comment by AndrewThien** [comment]
Hi there @erummas ,
Thanks for the response.
We found the problems and have been working on it since yesterday. This may take more time than we expected and we will let you know as soon as the solution comes up. Thanks for your patience.

**Comment by AndrewThien** [comment]
Hi @erummas,
We have good news on this bug: The fix has been found and we will probably deploy it by the next release early next week.

For reference, @spco, please also look at the issues caused by this bug here, in the earlier comments. Thanks

**Comment by erummas** [comment]
That's great to know @AndrewThien

**Comment by AndrewThien** [comment]
re-open issue until the fix has been deployed, tested and approved on the production

**Comment by AndyRae** [comment]
Should now be fixed in production.

**Comment by erummas** [comment]
Thanks @AndyRae, I am unable to access the mapper url though. 
![image](https://github.com/Health-Informatics-UoN/Carrot-Mapper/assets/58890026/62dbfbd8-8eee-4af1-9b74-7d3c77744ee5)

**Comment by AndyRae** [comment]
Thanks @erummas  - it seems to be up right now for me, I'm actively monitoring to see if this is an issue, or just a hiccup.

**Comment by erummas** [comment]
@AndyRae Thanks, after a few tries it's now up.

**Comment by erummas** [comment]
@AndyRae however, it is slow, the moment it went down for me I couldn't click on the table link, this time I am unable to access the scan report, it is stuck on processing status on the tab icon.

**Comment by AndyRae** [comment]
@erummas yes I think this is a wider issue, we're investigating.

**Comment by erummas** [comment]
@AndyRae strangely the mapper is going down randomly today, usually when I open a scan report ( opens fine initially before it stops responding, SHARE is the one I am working with). I am not adding any mappings today so that's not what is causing the issue. It does restore after sometime, not sure if this is automated restore or otherwise.

**Comment by AndyRae** [comment]
@erummas yes it's odd - maybe try using [carrot.ac.uk](https://carrot.ac.uk) instead. Your login/details will be the same.

**Comment by erummas** [comment]
@AndyRae now carrot.ac.uk isn't responding. I uploaded a new scan report https://carrot.ac.uk/scanreports/640/
with Dictionary. Mapped one table smr01_icd09.csv with automated mapping. However, could see that mapper didn't generate mappings for the other two tables smr01_icd10.csv and smr01_opcs4.csv. Thought I'd give it sometime and check again, when I checked again it stopped responding.

**Comment by AndrewThien** [comment]
Hi @erummas .
I wasn't checking on carrot.ac.uk but was trying adding some concepts on test.carrot.ac.uk. Everything works fine on my end.
Just asking to confirm, is adding concepts what you meant by "generate mappings"? Thanks!

**Comment by erummas** [comment]
@AndyRae yes generate mappings = automated concepts via dictionary, it went up for like 2 mins and is down again.
![image](https://github.com/Health-Informatics-UoN/Carrot-Mapper/assets/58890026/f0a57174-d30e-4c70-aa43-4585953c261e)

**Comment by AndyRae** [comment]
Hi @erummas - we've rolled back to using the "legacy" UI on both ccom and carrot.ac.uk, which should stop the downtime at least. Obviously this isn't ideal at all - so we'll be working to see what is going wrong.

**Comment by AndyRae** [comment]
Hi @erummas - we've reverted back to the "new" UI, hopefully with the fix in place, but we are actively monitoring for any downtime, and will rollback if there are any problems.

**Comment by erummas** [comment]
Thank you for the hard work @AndyRae and team. Fingers crossed, hope no issues this time

**Comment by AndyRae** [comment]
Looks like we're having trouble again, we're investigating and may have to roll back again.

**Comment by erummas** [comment]
@AndyRae just came here to report this. The mapper goes up but as soon as I try to access the scan report, it stops responding :(

**Comment by erummas** [comment]
It was working fine yesterday with the new interface.

**Comment by AndyRae** [comment]
Carrot looks to be currently down... but so is are many things using the same resources. We're investigating, but not sure this is Carrot related.

**Comment by AndyRae** [comment]
Carrot is currently having issues - again, this looks like it is an infrastructure issue and not Carrot itself.

**Comment by AndyRae** [comment]
We are seeing some down time this morning, we are investigating.

**Comment by AndyRae** [comment]
This has been fixed - the issue was that we were still maintaining the "legacy" UI, which forced the app to crash. We've turned it off for now, and maybe for good. So the app is slightly more resilient currently.

**Comment by erummas** [comment]
Good to know, thank you!

**Comment by erummas** [comment]
Hi Andy, Carrot is down. I am mapping SMR01 and there are many codes that have no automated mappings so I have to map them manually say W00.1,W00.2,W00.3,W00.4,W00.5,W00.6 and so on. Since the parent code W00 would map to all, I frequently use the search filter bar to find these codes so can be mapped to a single parent code at one time instead of browsing through 17000+ codes across100s of pages. I have observed that after entering the concept_id if the search bar is used before the concept appears, Carrot hangs up. I know I should wait till all concepts appear once I have entered them but anytime I forget to wait (sorry bad habit) it hangs. Not expecting for solution now but just making aware for future. For now could the mapper be restarted please? Thank you.

**Comment by AndyRae** [comment]
Hi @erummas - should be back up, but thanks so much. This is really helpful on what could be causing us problems, we can work to improve this!

**Comment by erummas** [comment]
Hi @AndyRae could you please get the Mapper back up -> I was using the search bar when it happened. I wasn't inserting any concept_ids  just looking for different codes in the data.

**Comment by AndyRae** [comment]
Should be back @erummas - we'll closely look at this urgently next week as well, sorry about htis.

**Comment by erummas** [comment]
No worries, thank you so much @AndyRae

**Comment by erummas** [comment]
Hi @AndyRae The mapper is down, could you please have a look?

**Comment by AndrewThien** [comment]
Hi @erummas , Carrot was just restarted. It should be up and running now. Please have a look. We are investigating this problem.

**Comment by erummas** [comment]
Hi @AndrewThien, thank you it's working now. No, I am afraid I do not. I logged in this morning to map a dataset and it wasn't working.

**Comment by AndyRae** [comment]
We're going to close this issue in favour of 

https://status.carrot.ac.uk

And all the active monitoring we have to support the uptime of Carrot now
